OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs

```diff
@@ -58,18 +63,94 @@ public void PlayMovieFullscreen(string movie, LuaFunction func = null)
 				};
 			}
 			else
-				onComplete = () => { };
+				onCompleteFullscreen = () => { };
 
-			Media.PlayFMVFullscreen(world, movie, onComplete);
+			Media.PlayFMVFullscreen(world, movie, onCompleteFullscreen);
 		}
 
-		[Desc("Display a text message to the player.")]
-		public void DisplayMessage(string text, string prefix = "Mission") // TODO: expose HSLColor to Lua and add as parameter
+		Action onLoadComplete;
+		Action onCompleteRadar;
+		[Desc("Play a VQA video in the radar window.  File name has to include the file extension. Returns true on success, if the movie wasn't found the function returns false and the callback is executed.")]
+		public bool PlayMovieInRadar(string movie, LuaFunction playComplete = null)
+		{
+			if (playComplete != null)
+			{
+				var f = playComplete.CopyReference() as LuaFunction;
+				onCompleteRadar = () =>
+				{
+					try
+					{
+						using (f)
+							f.Call();
+					}
+					catch (LuaException e)
+					{
+						Context.FatalError(e.Message);
+					}
+				};
+			}
+			else
+				onCompleteRadar = () => { };
+
+			Stream s = null;
+			try
+			{
+				s = GlobalFileSystem.Open(movie);
+			}
+			catch (FileNotFoundException e)
+			{
+				Log.Write("lua", "Couldn't play movie {0}! File doesn't exsist.", e.FileName);
```

Comment:
I disagree with this: people without the movies shouldn't be treated as second class, so the map should cope with the missing movies and do their own fallback instead of just relying on an ugly default error message.

There's also a typo here: exsist -> exist